src/js/core/directives/ui-grid-header-cell.js

```diff
@@ -338,21 +338,25 @@
 
             $scope.$on( '$destroy', dataChangeDereg );
 
-            $scope.handleClick = function(event) {
-              // If the shift key is being held down, add this column to the sort
-              var add = false;
-              if (event.shiftKey) {
-                add = true;
-              }
-
+            function applySort(add) {
               // Sort this column then rebuild the grid's rows
               uiGridCtrl.grid.sortColumn($scope.col, add)
                 .then(function () {
                   if (uiGridCtrl.columnMenuScope) { uiGridCtrl.columnMenuScope.hideMenu(); }
                   uiGridCtrl.grid.refresh();
                 }).catch(angular.noop);
+            }
+
+            $scope.handleClick = function(event) {
+              // If the shift key is being held down, add this column to the sort
+              applySort(event.shiftKey);
             };
 
+            $scope.handleKeyDown = function(event) {
+              if (event.keyCode === 32 || event.keyCode === 13) {
```

Comment:
keycode is deprecated in favor of key: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Reply:
updated to use key instead of keyCode since code does not have full support